scripts/bump-changelogs.sh

```diff
@@ -41,6 +41,10 @@ do
 done
 
 rm -rf $CHAP
-printf "\n!!!!!!\n%s %s\n!!!!!!\n" \
-  "WARNING! DO NOT BUMP THE VERSION NUMBER IN THE CABAL FILES" \
-  "(unless its dependencies were bumped)!"
+BUMPED=$(git diff --exit-code)
+if [[ $BUMPED ]]; then
+  printf "\n!!!!!!\n%s %s\n!!!!!!\n" \
+    "WARNING! DO NOT BUMP THE VERSION NUMBER IN THE CABAL FILES" \
+    "(unless its dependencies were bumped)!"
+fi
+echo $(( BUMPED ))
```

Comment:
If you're running `git diff --exit-code`, wouldn't it be better to check the exit code instead of the output of the command? But i guess it depends on what we're trying to do, which I don't exactly understand.  It should not  fail the CI if there are differences,  only show them?

Reply:
Oh, sorry, I missed your comment about the demonstration - so we do want to fail CI if there are differences.
Still, I think maybe `BUMPED=$(git diff --exit-code)` is a little dubious,  unless there is a good reason for it that escapes me.

Reply:
I return the exit code from the script, so it fails CI when it's non-zero. We do this in the `fourmolize` script as well, except I don't assign it to a variable there which is needed here because I don't want to show that `WARNING` text when there were no bumps. And absolutely, we want CI to fail if there are differences. Or rather, when there were bumps, which means that some packages were released so we need to bump the corresponding changelogs. Essentially, this works like the `fourmolize` script: we run it in CI with `git diff --exit-code` to make sure CI fails when changes are needed and when that's the case, we just run the script locally and commit the changes.

Reply:
Even so, capturing and checking the output when the condition can be obtained from the exit code seems clumsy. I think this is better:

```suggestion
if git diff --exit-code; then
  printf "\n!!!!!!\n%s %s\n!!!!!!\n" \
    "WARNING! DO NOT BUMP THE VERSION NUMBER IN THE CABAL FILES" \
    "(unless their dependencies were bumped)!"
fi
```

The diff output will still go to stdout, but the `if` uses the exit code.

I like scripts be machine-friendly and reusable, so in [format-changelogs.sh](https://github.com/IntersectMBO/cardano-ledger/blob/master/scripts/format-changelogs.sh) I chose to use `git diff -s --exit-code` to suppress the output and then used `git diff --exit-code` [in the workflow](https://github.com/IntersectMBO/cardano-ledger/blob/master/.github/workflows/haskell.yml#L667) after calling the script. This may be a bit extreme, though, since some people don't have a failed exit code reflected in their prompt (although we do have this in our `nix develop` shell). Some tools take a middle ground and output a message saying that some files were modified without showing all the diffs, and perhaps this would be a good compromise.